### PR TITLE
Updates refresh function

### PR DESCRIPTION
**Pull request by dylanlott**
First pass at a refresh function that PINGs nodes to check uptime. This only checks nodes that are in Kademlia. We should check all nodes that are in the cache, but I don't have a good way to grab the entire cache right now.

### CODE REVIEW

File: pkg/discovery/service.go [hunk 1]
```diff
@@ -39,16 +41,17 @@ func NewDiscovery(ol *overlay.Cache, kad *kademlia.Kademlia, stat statdb.DB) *Di
...
+	var errs []error
 	nodes := d.kad.Seen()
-
-	for _, v := range nodes {
-		if err := d.cache.Put(ctx, v.Id, *v); err != nil {
-			return err
+	for _, node := range nodes {
+		_, err := d.kad.Ping(ctx, *node)
+		if err != nil {
+			zap.L().Debug("error pinging node - discovery.Refresh", zap.Error(err))
```
Comment (wthorp) [inline pkg/discovery/service.go]: Should Discovery have its own logger instead?

Comment (dylanlott) [reply pkg/discovery/service.go]: Discovery probably should have it's own logger, you're right

File: pkg/discovery/service.go [hunk 2]
```diff
@@ -39,16 +41,17 @@ func NewDiscovery(ol *overlay.Cache, kad *kademlia.Kademlia, stat statdb.DB) *Di
 // We currently do not penalize nodes that are unresponsive,
 // but should in the future.
 func (d *Discovery) Refresh(ctx context.Context) error {
```
[the remark this answers is not in the document]
Comment (dylanlott) [reply pkg/discovery/service.go]: Good catch, I'll update that too

**Comment by jtolds** [comment]
there isn't really any benefit to checking all of the nodes in kad. i'd rather not merge this until we're actually only pinging every node in the cache. sounds like iterating through all the nodes in the cache is the next step

File: pkg/overlay/cache.go [hunk 3]
```diff
@@ -85,24 +95,25 @@ func (cache *Cache) GetAll(ctx context.Context, nodeIDs storj.NodeIDList) ([]*pb
...
+	// 		ns = append(ns, nil)
+	// 		continue
+	// 	}
+	// 	na := &pb.Node{}
+	// 	err := proto.Unmarshal(v, na)
+	// 	if err != nil {
+	// 		return nil, OverlayError.New("could not unmarshal non-nil node: %v", err)
+	// 	}
+	// 	ns = append(ns, na)
+	// }
```
Comment (jtolds) [inline pkg/overlay/cache.go]: please don't check in commented out code

**Pull request closed without merging**